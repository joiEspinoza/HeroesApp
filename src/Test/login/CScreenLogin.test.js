const { mount } = require("enzyme");
const { AuthContext } = require("../../Auth/AuthContext");
const { default: CScreenLogin } = require("../../Components/Login/CScreenLogin");

/////

describe('Pruebas en CScreenLogin', () => 
{
    const providerValue = { logged : false, dispatch : jest.fn() };
    const historyMock = { replace : jest.fn() };

    /////

    const wrapper = mount(

        <AuthContext.Provider value={ providerValue } >

            <CScreenLogin history={ historyMock } />

        </AuthContext.Provider>

    );  

   /////

   test('Debe mostrarse correctamente ', () => 
   {
        expect( wrapper ).toMatchSnapshot();
   });

    /////

   test('Debe realizazr dispatch y navegacion ', () => 
   {
       wrapper.find( "button" ).simulate( "click" );

       expect( providerValue.dispatch ).toHaveBeenCalledTimes( 1 );
       expect( historyMock.replace ).toHaveBeenCalledTimes( 1 );

       localStorage.setItem( "lastPath", "/dc" );

       wrapper.find( "button" ).simulate( "click" );

       expect( historyMock.replace ).toHaveBeenCalledWith( "/dc" );

   });

    
});
