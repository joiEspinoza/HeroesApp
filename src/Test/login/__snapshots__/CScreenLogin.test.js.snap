// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CScreenLogin Debe mostrarse correctamente  1`] = `
<div
  className="container mt-5"
>
  <h1>
    Screen | Login
  </h1>
  <hr />
  <button
    className="btn btn-outline-dark"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
